libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,164 @@
+// (C) 2022 GoodData Corporation
+import React from "react";
+import cx from "classnames";
+import { DateRangePickerInputField } from "./DateRangePickerInputField";
+import { injectIntl, WrappedComponentProps } from "react-intl";
+import { IntlWrapper } from "@gooddata/sdk-ui";
+import DayPickerInput from "react-day-picker/DayPickerInput";
+import moment from "moment";
+import { DateRangePickerInputFieldBody } from "./DateRangePickerInputFieldBody";
+import { convertPlatformDateStringToDate } from "../utils/DateConversions";
+import { TIME_FORMAT } from "../constants/Platform";
+
+interface IDateTimePickerOwnProps {
+    placeholderDate: string;
+    dateFormat: string;
+    onChange: (value: Date) => any;
+    value: Date;
+    dayPickerPropsWithDefaults: any;
+    handleDayClick: any;
+    locale: string;
+    isMobile: boolean;
+    isTimeForStaticPeriodEnabled: boolean;
+    className: string;
+    defaultTime?: string;
+    error?: boolean;
+}
+
+export type DateTimePickerComponentProps = IDateTimePickerOwnProps & WrappedComponentProps;
+
+// eslint-disable-next-line react/display-name
+export const DateTimePickerComponent = React.forwardRef<DayPickerInput, DateTimePickerComponentProps>(
+    (props: DateTimePickerComponentProps, ref) => {
+        const {
+            placeholderDate,
+            value,
+            onChange,
+            dateFormat,
+            dayPickerPropsWithDefaults,
+            handleDayClick,
+            isMobile,
+            isTimeForStaticPeriodEnabled,
+            className,
+            defaultTime = "00:00",
+            error = false,
+        } = props;
+
+        const getFormattedDate = () => {
```

Comment:
I would move this function outside of the component and provide `value` as a parameter.

Reply:
I got your point but I would rather keep it as a part of the component since it is related to internal representation of the Calendar component within datetimePicker. It can change with a new calendar component. I changed the name of the method to getDate if that helps.

Reply:
I'm rather if component body it as short as possible and easy to grasp what it does and have logic such as this extracted to "static" functions outside of the component function or into a hook. Often times it is easier to test even. However, leave it as is. It was just a suggestion.